ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/TerminateTransientQueryFunctionalTest.java

```diff
@@ -0,0 +1,153 @@
+  private static final TestKsqlRestApp REST_APP_0 = TestKsqlRestApp
+      .builder(TEST_HARNESS::kafkaBootstrapServers)
+      .withStaticServiceContext(TEST_HARNESS::getServiceContext)
+      .withProperty(KsqlRestConfig.LISTENERS_CONFIG, "http://localhost:8088")
+      .withProperty(KsqlRestConfig.ADVERTISED_LISTENER_CONFIG, "http://localhost:8088")
+      .build();
+  private static final TestKsqlRestApp REST_APP_1 = TestKsqlRestApp
+      .builder(TEST_HARNESS::kafkaBootstrapServers)
+      .withStaticServiceContext(TEST_HARNESS::getServiceContext)
+      .withProperty(KsqlRestConfig.LISTENERS_CONFIG, "http://localhost:8089")
+      .withProperty(KsqlRestConfig.ADVERTISED_LISTENER_CONFIG, "http://localhost:8089")
+      .build();
+
+  @ClassRule
+  public static final RuleChain CHAIN = RuleChain
+      .outerRule(Retry.of(3, ZooKeeperClientException.class, 3, TimeUnit.SECONDS))
+      .around(TEST_HARNESS)
+      .around(REST_APP_0)
+      .around(REST_APP_1);
+  @BeforeClass
+  public static void setUpClass() {
+    TEST_HARNESS.ensureTopics(PAGE_VIEW_TOPIC);
+    RestIntegrationTestUtil.createStream(REST_APP_0, PAGE_VIEWS_PROVIDER);
+    RestIntegrationTestUtil.makeKsqlRequest(
+        REST_APP_0,
+        "CREATE STREAM S AS SELECT * FROM " + PAGE_VIEW_STREAM + ";"
+    );
+  }
+
+  @Test
+  public void shouldTerminatePushQueryOnSameNode() {
+    // Given:
+    givenPushQuery();
+    final String transientQueryId = getTransientQueryIds().get(0);
+
+    // When:
+    RestIntegrationTestUtil.makeKsqlRequest(
+        REST_APP_0,
+        "terminate " + transientQueryId + ";"
+    );
+
+    // Then:
+    assertThat(
+        "Should terminate push query on same node using query id",
+        !checkForTransientQuery()
+    );
+  }
+
+  @Test
+  public void shouldTerminatePushQueryOnAnotherNode() {
+    // Given:
+    givenPushQuery();
+    final String transientQueryId = getTransientQueryIds().get(0);
+
+    // When:
+    RestIntegrationTestUtil.makeKsqlRequest(
+        REST_APP_1,
+        "terminate " + transientQueryId + ";"
+    );
+
+    // Then:
+    assertThat(
+        "Should terminate push query on another node using query id",
+        !checkForTransientQuery()
+    );
+  }
+
+  public List<RunningQuery> showQueries (){
+    return ((Queries) RestIntegrationTestUtil.makeKsqlRequest(
+        REST_APP_0,
+        "show queries;"
+    ).get(0)).getQueries();
+  }
+
+  public boolean checkForTransientQuery (){
+    return showQueries().stream()
+        .anyMatch(q -> q.getId().toString().contains("transient"));
+  }
+
+  public List<String> getTransientQueryIds () {
+    return showQueries().stream()
+        .filter(q -> q.getId().toString().contains("transient"))
+        .map(q -> q.getId().toString())
+        .collect(Collectors.toList());
+  }
+
+  public void givenPushQuery() {
+    ExecutorService service = Executors.newFixedThreadPool(2);
```

Comment:
We should try to set up the executor service in setup and make sure we shut it down in teardown using `shutdownNow()`.  That way we don't leave threads hanging around while running tests and slowing things down.  Also, it seems this needs just one thread.